server/lib/bitbucketApi.js

```diff
@@ -21,7 +22,42 @@ export default class Bitbucket {
       }
     }, options);
 
-    this.request = request;
+    this.request = (opts, cb) => {
+      request(opts, (error, response, data) => {
+        if (error) {
+          cb(error);
+        } else {
+          try {
+            data = JSON.parse(data); // eslint-disable-line no-param-reassign
+            if (typeof data.errors !== 'undefined') {
+              cb(data.errors);
+            } else if (response.statusCode !== 200) {
+              cb(this.generateApiError(response.statusCode, response));
+            } else {
+              cb(null, data);
+            }
+          } catch (parseError) {
+            if (response.statusCode === 200) {
+              cb(null, data);
+            } else {
+              cb(this.generateApiError(response.statusCode, response));
+            }
+          }
+        }
+      }).auth(this.options.user_name, this.options.password, true);
+    };
+
+    this.generateApiError = (statusCode, response = null) => {
+      const error = new Error(`Error ${statusCode} when calling '${options.method.toUpperCase()} ${options.url}' (username: ${this.options.user_name})`);
```

Comment:
Now that we've moved the `generateApiError` to the constructor from `doRequest` we no longer have access to the `options` object and `options.method.toUpperCase()` results in an `uncaughtException`.

It leaves the spinner running indefinitely when the deploy fails due to this uncaught